### PR TITLE
feat: track nonce

### PR DESCRIPTION
**Pull request by justprosh**

### CODE REVIEW

File: crates/chain-connector/src/connector.rs [hunk 1]
```diff
@@ -487,12 +487,25 @@ impl HttpChainConnector {
...
                 .await,
-        )?;
-        Ok(resp)
+        );
+
+        match result {
+            Ok(resp) => {
+                let new_nonce = nonce + Uint::from(1);
+                *nonce_guard = Some(new_nonce);
+                tracing::debug!(target: "chain-connector", "Incrementing nonce. New nonce {new_nonce}");
```
Comment (folex) [inline crates/chain-connector/src/connector.rs:500]: maybe move to info for now?

or better ask providers to enable debug for chain connector?

File: crates/chain-connector/src/connector.rs [hunk 2]
```diff
@@ -82,7 +79,7 @@ pub trait ChainConnector: Send + Sync {
 pub struct HttpChainConnector {
     client: Arc<jsonrpsee::http_client::HttpClient>,
     config: ChainConfig,
-    tx_nonce_mutex: Arc<Mutex<()>>,
+    tx_nonce_mutex: Arc<Mutex<Option<U256>>>,
```
Comment (folex) [inline crates/chain-connector/src/connector.rs:82]: i think you could be fine with atomic u64, no mutex and probably no error/result

**Review by folex — APPROVED** [review]

File: crates/chain-connector/src/connector.rs (hunk 2 above)
Comment (justprosh) [reply crates/chain-connector/src/connector.rs:82]: Nope, I need mutex to send only one tx at a time. In theory decider can send txs at the same time

File: crates/chain-connector/src/connector.rs (hunk 1 above)
Comment (justprosh) [reply crates/chain-connector/src/connector.rs:500]: I think better to ask for debug for now, I don't like to make it info temporary

**Pull request merged**